### PR TITLE
User Interface Header Files

### PR DESCRIPTION
**Pull request by Andrew-Ritchie**
# Details

- Added header file for the main window.
- Added header file for the full-screen live feed.
- Added header file for the settings page.
- Added header file for the data page.

# Checklist

## Code
- [ ] I have added some tests
- [ ] I have run the tests locally to ensure they all pass

## Documentation
- [ ] I have updated documentation appropriately (GitHub Pages)
- [ ] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)

Closes Issue #

### CODE REVIEW

File: src/gui/interface/Header Files/datapage.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef SRC_DATAPAGE_H_
+#define SRC_DATAPAGE_H_
+
+#include <QMainWindow>
+
+namespace Ui {
```
Comment (C-Begley) [inline src/gui/interface/Header Files/datapage.h]: Would UI look better?

File: src/gui/interface/Header Files/fullscreen.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ */
+
+#ifndef SRC_FULLSCREEN_H_
+#define SRC_FULLSCREEN_H_
+
+#include <QMainWindow>
+
+
+namespace Ui {
```
Comment (C-Begley) [inline src/gui/interface/Header Files/fullscreen.h]: How does namespacing work on multiple headers?

File: src/gui/interface/Header Files/settingspage.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+class settingsPage : public QMainWindow{
+    Q_OBJECT
+
+ public:
+    /*
+      * @brief Initialises the settings page.
+      */
+    explicit settingsPage(QWidget *parent = nullptr);
```
Comment (C-Begley) [inline src/gui/interface/Header Files/settingspage.h:47]: Does this need a @param field?
And the others

File: src/gui/interface/Header Files/fullscreen.h [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    void on_settings_button_clicked();
+
+
+
+
+ private:
+    /*
+      * @brief Closes the full screen page.
+      */
+    Ui::FullScreen *ui;
```
Comment (C-Begley) [inline src/gui/interface/Header Files/fullscreen.h]: I think namespaces are supposed to be unique from the objects inside. But I think that is just a personal preference thing

File: src/gui/interface/Header Files/datapage.h (hunk 1 above)
Comment (miklasr) [reply src/gui/interface/Header Files/datapage.h]: It probably would look better, but I guess `Ui` matches the "official" naming scheme

Comment (Andrew-Ritchie) [reply src/gui/interface/Header Files/datapage.h]: I have used GUI now to fix this and other comments! :)

File: src/gui/interface/Header Files/settingspage.h (hunk 3 above)
Comment (Andrew-Ritchie) [reply src/gui/interface/Header Files/settingspage.h:47]: I have added it in now. :)

File: src/gui/interface/Header Files/fullscreen.h (hunk 4 above)
Comment (Andrew-Ritchie) [reply src/gui/interface/Header Files/fullscreen.h]: I have changed it to GUI to fix this issue. :)

**Review by Ashwin-MJ — APPROVED** [review]
Looks good to me! Nice stuff 😄

**Review by miklasr — APPROVED** [review]

**Review by C-Begley — APPROVED** [review]
I'm happy with changes

**Pull request merged**